lib/Doctrine/DBAL/Schema/SQLServerSchemaManager.php

```diff
@@ -37,6 +37,31 @@ class SQLServerSchemaManager extends AbstractSchemaManager
     /**
      * {@inheritdoc}
      */
+    public function dropDatabase($database)
+    {
+        try {
+            parent::dropDatabase($database);
+        } catch (DBALException $exception) {
+            /** @var \Doctrine\DBAL\Driver\DriverException $exception */
+            $exception = $exception->getPrevious();
+
+            // If we have a error code 3702, the drop database operation failed
+            // because of active connections on the database.
+            // To force dropping the database, we first have to close all active connections
+            // on that database and issue the drop database operation again.
+            if ($exception->getErrorCode() !== 3702) {
```

Comment:
I disagree here. It doesn't help anyone and if we start to introduce constants for all error codes and SQL states, we will go crazy. We started with this approach when we implemented driver exception converters and discarded that approach again. This check will disappear as soon as we implement SQL Server driver as driver exception converter drivers (hopefully) as we will be able to catch a dedicated exception type then. Also it is implemented that way for PostgreSQL already and I don't want to do everything that is necessary to make it "perfect" with this PR. First I want to get the SQL Server test suite to be green. We can still improve afterwards. What do you think?